### ISSUE TITLE
Thoughts after the.NET AI presentation

### ISSUE THREAD

**Issue by whuanle** [xamarin/TorchSharp#1467]
On March 22, 2025, I gave a talk on.NET AI in Shenzhen, China, where I explained a bit about TorchSharp and ML.NET.
<br />
Event information:  
[https://globalai.community/bootcamp/china-shenzhen/](https://globalai.community/bootcamp/china-shenzhen/)
<br />
Since PPT can only hold a little content, I created a separate Github warehouse to demonstrate cases and explain the detailed process.
[https://github.com/whuanle/20250322](https://github.com/whuanle/20250322)

<br />
I have a lot of readers in China who are also interested, in the Internet of Things and various fields, many developers also want to be able to achieve AI model training and reasoning through the.NET platform.
<br />
I think there are a couple of things that need to happen right now to really be able to train and deploy AI through the .NET platform.  

* TorchSharp continues to synchronize some of Pytorch's network structures, for example.  
* ML.NET is combined with TorchSharp, and the AI models trained by TorchSharp can be directly converted to.onnx format.  

![Image](https://github.com/user-attachments/assets/06932eee-971e-4f58-93b8-39753fb3d2f8)

If I can do these two things, I can quickly implement more cases, use them in enterprise projects, and publicize and demonstrate them to readers.

**Comment by GeorgeS2019** [comment]
Promote directly at the Torchsharp discussion...tell people what U have learned and what are needed next

**Comment by whuanle** [comment]
@GeorgeS2019 
I think the dotnet/TorchSharp project is at a standstill...

**Comment by GeorgeS2019** [comment]
There are 3 Microsoft engineers dedicated to keep it going...strategic to ML.net and agent Ai for Dotnet.

Please keep pushing and ask for feedback...perhaps the 3 engineers can coordinate with what U hope to achieve

**Comment by GeorgeS2019** [comment]
Currently, it is unclear if the existing implementation of TorchVision requires the [c++ part of the PyTorch Vision. ](https://github.com/pytorch/vision/tree/main/torchvision/csrc)Anyone know this answer? 

# TorchSharp TorchVision status
![Image](https://github.com/user-attachments/assets/4dc02e16-ac9c-48f1-b5df-0abf1ebfafd1)
# PyTorch Vision
![Image](https://github.com/user-attachments/assets/7ce9ccfe-b7c0-4931-8135-5ad872a4197c)

**Comment by GeorgeS2019** [comment]
Hi @ozanMSFT 

###part1

> If you're looking for the examples, I suggest you to visit:
https://github.com/dotnet/TorchSharpExamples

@whuanle is not looking for example...in contrary..he is addressing important missing part of the Torchsharp ecosystem.

Microsoft learn offers fundamental python with pytorch lessons as one of AI modules.

@whuanle is attempting to port the pytorch lessons to Torchsharp...so in future, students could choose to learn AI through pytorch or TorchSharp.

### part 2

> If you're having an issue regarding the missing implementation, could you please create an issue by specifying that functionality?

He noticed that the current implementation of the model part of  torchvision c# is only 10% of what is available in Pytorch vision.

This prevents him from completing the part 1: the Microsoft learn AI through TorchSharp

**Comment by GeorgeS2019** [comment]
@ozanMSFT 

Perhaps start introducing label: TorchVision, TorchAudio
